src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -871,38 +903,49 @@ void RNN::createDescriptor(const std::vector<MemoryDescPtr> &inputDesc,
+
+        attr->set_rnn_weights_qparams(weightsScaleMask, weightsScales);
+        attr->set_rnn_data_qparams(inputScale, inputShift);
+    }
+
+    return attr;
+}
+
 void RNN::prepareParams() {
     for (size_t i = 0; i < wIdx; i++) {
         auto memPtr = getParentEdgesAtPort(i).front()->getMemoryPtr();
         if (!memPtr || !memPtr->isAllocated())
             THROW_ERROR << "has uninitialized memory at port " << i;
     }
 
-    const auto& dataPrecision = getOriginalInputPrecisionAtPort(0);
-    const auto dataType = DnnlExtensionUtils::IEPrecisionToDataType(dataPrecision);
-
     auto dataMemPtr = getParentEdgesAtPort(0).front()->getMemoryPtr();
     const size_t B = dataMemPtr->GetShape().getStaticDims()[0];
     const size_t SL = is_cell ? 1lu : dataMemPtr->GetShape().getStaticDims()[1];
     const Shape shapeS_4D{L, D, B, SC};
 
-    inDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, DC}, dataType, memory::format_tag::tnc);
-    outDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, SC}, dataType, memory::format_tag::tnc);
+    inDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, DC}, inDataTypes[xIdx], memory::format_tag::tnc);
+    outDataDescs[0] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, D * SC}, outDataTypes[yIdx], memory::format_tag::tnc);
 
-    inDataDescs[1] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, dataType, memory::format_tag::ldnc);
-    outDataDescs[1] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, dataType, memory::format_tag::ldnc);
+    inDataDescs[1] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, inDataTypes[hIdx], memory::format_tag::ldnc);
+    outDataDescs[1] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, outDataTypes[hoIdx], memory::format_tag::ldnc);
 
     if (haveCellState(cell_type)) {
-        inDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, memory::data_type::f32, memory::format_tag::ldnc);
-        outDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, memory::data_type::f32, memory::format_tag::ldnc);
+        inDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, inDataTypes[cIdx], memory::format_tag::ldnc);
+        outDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(shapeS_4D, outDataTypes[coIdx], memory::format_tag::ldnc);
+    } else if (haveAttention(cell_type)) {
+        inDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, 1}, inDataTypes[aIdx], memory::format_tag::tnc);
     }
 
-    if (haveAttention(cell_type)) {
-        inDataDescs[2] = std::make_shared<DnnlBlockedMemoryDesc>(Shape{SL, B, 1}, dataType, memory::format_tag::tnc);
-    }
     bool wFormatWasChanged = false;
     // WA To avoid different weights layer and iter formats in FP32 case.
-    if ((dataPrecision == Precision::FP32) && (SL != 1 || B < optimalBatchSize)) {
+    if (one_of(inDataTypes[xIdx], memory::data_type::f32, memory::data_type::bf16) &&
```

Comment:
@EgorDuplensky @usstq Here is a concern -
1. In situation of AUGRU and GRU with DIEN, it is measured and confirmed that -
in BF16, we should use format_tag::any and then onednn will choose amx_brgemm gru path to get best performance.
in FP32, we should use format_tag::ldigo instead to get best performance.

2. meanwhile, in situation of LSTM with RNNT, it is measure and confirmed that -
In both case of FP32 and BF16, format_tag::any should be used to get best performance.

Reply:
Reverted in subsequent PR